vpc/subnets/main.tf

```diff
@@ -0,0 +1,53 @@
+# Find the available availability zones
+data "aws_availability_zones" "available" {}
+
+# Sets up the public subnet's route table
+resource "aws_route_table" "mod_public" {
+  vpc_id = "${var.vpc_id}"
+  route {
+      cidr_block = "0.0.0.0/0"
+      gateway_id = "${var.internet_gateway_id}"
+  }
+  tags { Name = "${var.environment_name}-public" }
+}
+
+# Sets up the private subnet's route table
+# Send all traffic over the nat gateway
+resource "aws_route_table" "mod_private" {
+  vpc_id = "${var.vpc_id}"
+  tags { Name = "${var.environment_name}-private" }
+}
+
+# Sets up the private subnets
+resource "aws_subnet" "mod_private" {
+  vpc_id = "${var.vpc_id}"
+  cidr_block = "${cidrsubnet(var.vpc_cidr_block, var.private_newbits, var.private_netnum_offset + lookup(var.az_to_netnum, substr(element(data.aws_availability_zones.available.names, count.index), -1, 1)))}"
```

Comment:
can we do some of this variable assignment on multiple lines? Might make it more readable/understandable.

Reply:
My thoughts exactly! I was thinking about moving this to a template_file.

Reply:
https://www.terraform.io/docs/configuration/locals.html might help here too

Reply:
I tried locals initially, but needing to iterate over the entire map makes it tough.